dace/sourcemap.py

```diff
@@ -61,45 +68,34 @@ def get_tmp(name):
     else:
         return None
 
-def remove_tmp(name, remove_cache=False):
-    ''' Remove the tmp file of the function 'name'.
-        If remove_cache is true then check if the 
-        directory only contains the tmp file,
-        if this is the case, remove the cache directory
-        of this function.
-    '''
+
+def remove_tmp(name: str, remove_cache: bool = False):
+    """ Remove the tmp created by "temporaryInfo"
+        :param name: name of the function whiches tmp will be removed
+        :param remove_cache: If true, checks if the directory only contains 
+        the tmp file, if this is the case, remove the SDFG cache directory.
+    """
     path = ".dacecache/" + name
 
     if not os.path.exists(path):
         return
 
-    try:
-        os.remove(path + "/map/tmp.json")
-    except OSError:
-        print("removing error 1")
-        return
-    
-    if (
-        remove_cache and
-        len(os.listdir(path)) == 1 and
-        len(os.listdir(path + "/map")) == 0
-    ):
+    os.remove(path + "/map/tmp.json")
```

Comment:
for compatibility, please use 
```suggestion
    os.remove(os.path.join(path, 'map', 'tmp.json'))
```